apps/waterfall/waterfall/src/app/dashboard/title/create/create.component.ts

```diff
@@ -0,0 +1,84 @@
+// Angular
+import { ActivatedRoute, Router } from '@angular/router';
+import { FormControl, Validators } from '@angular/forms';
+import { Component, ChangeDetectionStrategy, OnInit } from '@angular/core';
+import { BehaviorSubject, Observable, combineLatest, map, startWith } from 'rxjs';
+
+// Blockframes
+import { MovieService } from '@blockframes/movie/service';
+import { MovieForm } from '@blockframes/movie/form/movie.form';
+import { RightholderRole, createAppConfig } from '@blockframes/model';
+import { OrganizationService } from '@blockframes/organization/service';
+import { WaterfallService } from '@blockframes/waterfall/waterfall.service';
+import { FileUploaderService } from '@blockframes/media/file-uploader.service';
+import { WaterfallPermissionsService } from '@blockframes/waterfall/permissions.service';
+
+
+@Component({
+  selector: 'waterfall-title-create',
+  templateUrl: './create.component.html',
+  styleUrls: ['./create.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CreateComponent implements OnInit {
+
+  movieForm = new MovieForm({ directors: [{ firstName: '', lastName: '' }] });
+
+  waterfallForm = new FormControl<RightholderRole[]>(undefined, [Validators.required]);
+
+  movieId = this.movieService.createId();
+
+  invalid$: Observable<boolean>;
```

Comment:
you can do this to remove ngOnInit
```suggestion
  invalid$ = combineLatest([
      this.movieForm.valueChanges,
      this.waterfallForm.valueChanges,
    ]).pipe(
      map(([ movie, waterfall ]) => {
        // check movie
        if (!movie.title.international) return true;
        if (movie.directors.length === 0) return true;
        const missing = movie.directors.some(d => !d.firstName || !d.lastName);
        if (missing) return true;

        // check waterfall
        if (waterfall.length === 0) return true;

        return false;
      }),
      startWith(true),
    );
```